lib/srv/reexec.go

```diff
@@ -612,9 +612,96 @@ func (o *osWrapper) startNewParker(ctx context.Context, credential *syscall.Cred
 	return nil
 }
 
-// RunForward reads in the command to run from the parent process (over a
+type forwardHandlerParams struct {
+	addr           string
+	fileDescriptor *os.File
+}
+
+type forwardHandler func(ctx context.Context, params forwardHandlerParams) error
```

Comment:
A signature like this would make things easier to follow IMO.

```suggestion
type forwardHandler func(ctx context.Context, payload string, file *os.File) error
```